pkgs/applications/misc/latte-dock/default.nix

```diff
@@ -1,30 +1,32 @@
 { mkDerivation, lib, cmake, xorg, plasma-framework, plasma-wayland-protocols, fetchFromGitLab
-, extra-cmake-modules, karchive, kwindowsystem, qtx11extras, qtwayland, kcrash, knewstuff, wayland }:
+, extra-cmake-modules, karchive, kwindowsystem, qtx11extras, qtwayland, kcrash, knewstuff, wayland, kde-cli-tools, makeWrapper }:
 
 mkDerivation rec {
   pname = "latte-dock";
-  version = "unstable-2022-09-06";
+  version = "unstable-2023-01-26";
 
   src = fetchFromGitLab {
     domain = "invent.kde.org";
     owner = "plasma";
     repo = "latte-dock";
-    rev = "cd36798a61a37652eb549d7dfcdf06d2028eddc4";
-    sha256 = "sha256-X2PzI2XJje4DpPh7gTtYnMIwerR1IDY53HImvEtFmF4=";
+    rev = "159496edc0609c953a813a41844bbb4f0ee22db4";
+    sha256 = "sha256-WqAlnwD/Q7+OUf2MuJfsFkruYQnlYahRCm1p+1zu8eA=";
   };
 
-  buildInputs = [ plasma-framework plasma-wayland-protocols qtwayland xorg.libpthreadstubs xorg.libXdmcp xorg.libSM wayland ];
+  buildInputs = [ plasma-framework plasma-wayland-protocols qtwayland xorg.libpthreadstubs xorg.libXdmcp xorg.libSM wayland kde-cli-tools ];
 
   nativeBuildInputs = [ extra-cmake-modules cmake karchive kwindowsystem
-    qtx11extras kcrash knewstuff ];
+    qtx11extras kcrash knewstuff makeWrapper ];
 
   patches = [
     ./0001-Disable-autostart.patch
+    ./0002-Detect-kde-version.patch
   ];
 
   postInstall = ''
     mkdir -p $out/etc/xdg/autostart
     cp $out/share/applications/org.kde.latte-dock.desktop $out/etc/xdg/autostart
+    wrapProgram $out/bin/latte-dock --prefix PATH : ${lib.makeBinPath [ kde-cli-tools ]}
```

Comment:
Do I understand correctly that this is newly required? (i.e. that this worked beforehand and something changed in latte-dock that caused it to start using kde-cli-tools in the meantime)

Reply:
If you install the latte-dock version currently available in the Nix repo, you will see that clicking on an app icon in the dock will not do anything, which is kind of a core feature that it should open an app or show the window when clicking on an app. I described the issue here: https://github.com/NixOS/nixpkgs/issues/212828. The problem is that latte-dock is trying to detect the plasmashell version by parsing the output of `plasmashell -v` at run time. I tried to wrap plasma-workspace with latte-dock but to not avail, I think due to plasmashell segfaulting on nixos:
```
[aroun@nixos:~]$ plasmashell -v
plasmashell 5.26.4
Segmentation fault (core dumped)
```
So I wrapped kde-cli-tools instead, and added the /0002-Detect-kde-version.patch to replace `plasmashell -v` by `kbroadcastnotification -v`, which does not segfault and results in fixing the issue of apps not opening. I also thought that maybe wrapping kde-cli-tools would be lighter than wrapping plasma-workspace.
```
[:~] % kbroadcastnotification -v
kbroadcastnotification 5.26.4
[:~] %
```

Reply:
Why was it checking `plasmashell` in the first place? Do you think someone just picked a random KDE thing, or is there a reason why they picked `plasmashell` and not e.g. `kbroadcastnotification`?